hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/DataStreamer.java

```diff
@@ -698,11 +698,14 @@ public void run() {
               LOG.debug("Thread interrupted", e);
             }
             one = dataQueue.getFirst(); // regular data packet
-            SpanId[] parents = one.getTraceParents();
+            SpanContext[] parents = one.getTraceParents();
             if (parents.length > 0) {
+              // The original code stored multiple parents in the DFSPacket, and
+              // use them ALL here when creating a new Span. We only use the
+              // last one FOR NOW. Moreover, we don't activate the Span for now.
               scope = dfsClient.getTracer().
-                  newScope("dataStreamer", parents[0]);
-              scope.getSpan().setParents(parents);
+                  newScope("dataStreamer", parents[0], false);
+              //scope.getSpan().setParents(parents);
```

Comment:
i need to check if the concept matches but io.opentracing.Tracer offers a addReference() API to represent the casual relationship.